README.md

```diff
@@ -1,44 +1,153 @@
+
+ A | B | C | D |NOT C|NOT A|NOT A AND D|B OR NOT C|NOT (B OR NOT C)|A AND NOT (B OR NOT C)|A AND NOT (B OR NOT C) AND (NOT A AND D)
+---|---|---|---|---|---|---|---|---|---|---
+ 0 | 0 | 0 | 0 | 1 | 1 | 0 | 1 | 0 | 0 | 0
+ 0 | 0 | 0 | 1 | 1 | 1 | 1 | 1 | 0 | 0 | 0
+ 0 | 0 | 1 | 0 | 0 | 1 | 0 | 0 | 1 | 0 | 0 
+ 0 | 0 | 1 | 1 | 0 | 1 | 1 | 0 | 1 | 0 | 0 
+ 0 | 1 | 0 | 0 | 1 | 1 | 0 | 1 | 0 | 0 | 0 
+ 0 | 1 | 0 | 1 | 1 | 1 | 1 | 1 | 0 | 0 | 0 
+ 0 | 1 | 1 | 0 | 0 | 1 | 0 | 1 | 0 | 0 | 0 
+ 0 | 1 | 1 | 1 | 0 | 1 | 1 | 1 | 0 | 0 | 0 
+ 1 | 0 | 0 | 0 | 1 | 0 | 0 | 1 | 0 | 0 | 0 
+ 1 | 0 | 0 | 1 | 1 | 0 | 0 | 1 | 0 | 0 | 0 
+ 1 | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 1 | 1 | 0 
+ 1 | 0 | 1 | 1 | 0 | 0 | 0 | 0 | 1 | 1 | 0 
+ 1 | 1 | 0 | 0 | 1 | 0 | 0 | 1 | 0 | 0 | 0 
+ 1 | 1 | 0 | 1 | 1 | 0 | 0 | 1 | 0 | 0 | 0 
+ 1 | 1 | 1 | 0 | 0 | 0 | 0 | 1 | 0 | 0 | 0 
+ 1 | 1 | 1 | 1 | 0 | 0 | 0 | 1 | 0 | 0 | 0 
 
-### b
 
-### c
+## Question 2
 
-### d
+### A: A AND B AND NOT C
 
-## Question 2
+![1A](1A.png)
 
-### a
+### B: A AND NOT (B AND NOT C)
 
-### b
+![1B](1B.png)
 
-### c
+### C: (A OR NOT B)  AND  (A OR C)
 
-### d
+![1C](1C.png)
 
-## Question 3
+### D: A AND NOT (B OR NOT C) AND (NOT A AND D)
 
-### a
+![1D](1D.png)
 
-### b
 
-### c
+## Question 3
 
-### d
+### A: NOT (A OR B) = NOT A AND NOT B
+
+ A | B | A OR B | NOT A | NOT B | NOT (A OR B) | NOT A AND NOT B
+---|---|---|---|---|---|---
+ 0 | 0 | 0 | 1 | 1 | 1 | 1 
+ 0 | 1 | 1 | 1 | 0 | 0 | 0 
+ 1 | 0 | 1 | 0 | 1 | 0 | 0 
+ 1 | 1 | 1 | 0 | 0 | 0 | 0 
+ 
+### B: NOT (A AND B) = NOT A OR NOT B
+
+ A | B | A AND B | NOT A | NOT B | NOT (A AND B) | NOT A OR NOT B
+---|---|---|---|---|---|---
+ 0 | 0 | 0 | 1 | 1 | 1 | 1 
+ 0 | 1 | 0 | 1 | 0 | 1 | 1 
+ 1 | 0 | 0 | 0 | 1 | 1 | 1 
+ 1 | 0 | 0 | 0 | 1 | 1 | 1 
+ 1 | 1 | 1 | 0 | 0 | 0 | 0 
+
+### C: (A AND B) OR (A AND C) = A AND (B OR C)
+
+ A | B | C | A AND B | A AND C | B OR C | (A AND B) OR (A AND C) | A AND (B OR C)
+---|---|---|---|---|---|---|---
+ 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 
+ 0 | 0 | 1 | 0 | 0 | 1 | 0 | 0 
+ 0 | 1 | 0 | 0 | 0 | 1 | 0 | 0 
+ 0 | 1 | 1 | 0 | 0 | 1 | 0 | 0 
+ 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 
+ 1 | 0 | 1 | 0 | 1 | 1 | 1 | 1 
+ 1 | 1 | 0 | 1 | 0 | 1 | 1 | 1 
+ 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 
+
+### D: (A OR B) AND (A OR C) = A OR (B AND C)
+
+ A | B | C | A OR B | A OR C | B AND C | (A OR B) AND (A OR C) | A OR (B AND C)
+---|---|---|---|---|---|---|---
+ 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 
+ 0 | 0 | 1 | 0 | 1 | 0 | 0 | 0 
+ 0 | 1 | 0 | 1 | 0 | 0 | 0 | 0 
+ 0 | 1 | 1 | 1 | 1 | 1 | 1 | 1 
+ 1 | 0 | 0 | 1 | 1 | 0 | 1 | 1 
+ 1 | 0 | 1 | 1 | 1 | 0 | 1 | 1 
+ 1 | 1 | 0 | 1 | 1 | 0 | 1 | 1 
+ 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1
 
 ## Question 4
 
-### a
+### A
+
+It is the same as Question 3B.
+
+The line of code `if not(file_exists("a.txt") and file_exists("b.txt"))`\
+Is the same as `NOT (A AND B)`
+
+And the line of code `if not file_exists("a.txt") or not file_exists("b.txt")`\
+Is the same as `NOT A OR NOT B`
+
+`NOT (A AND B)` is equal to `NOT A OR NOT B`
```

Comment:
✔ Good reasoning -- try applying this kind of thinking to the rest of the parts of q4